### PR TITLE
add checker board for clip_path,clip_rect,clip_rrect,physical_shap_layer

### PR DESCRIPTION
**Pull request by fredlee12345678**
## Description
Some layer not support checkerboardOffscreenLayers diagnostic flag even set checkerboardOffscreenLayers to true.
So I just add some checkerboard to the layers.

## Related Issues

https://github.com/flutter/flutter/issues/11332
## Tests

I added the following tests:

checked: ClipRRect, ClipRect. ClipPathLayer, physical shape layer using demo to check if it's available

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x ] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x ] I signed the [CLA].
- [x ] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x ] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x ] I updated/added relevant documentation.
- [x ] All existing and new tests are passing.
- [ x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by fredlee12345678** [comment]
@iskakaushik I have updated test case, could you pls review my code when you are free?

**Review by iskakaushik — COMMENTED** [review]
I'd like to see more tests for this. Specifically asserting that checkerboard is indeed visible in these cases when enabled. Is it possible to write a driver test in github.com/flutter/flutter repo?

File: flow/layers/clip_path_layer.cc [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "flutter/flow/layers/clip_path_layer.h"
+#include <flutter/flow/paint_utils.h>
```
[the remark this answers is not in the document]
Comment (fredlee12345678) [reply flow/layers/clip_path_layer.cc]: fixed

**Comment by fredlee12345678** [comment]
@iskakaushik  check board using random color, so how should I write a test to detect check board is visible? could you please  give me some sample code?

**Comment by fredlee12345678** [comment]
@iskakaushik I have add unit test  in clip_rect_layer_unittests.cc, could you pls help to review it? if you think the test case is ok, I will finish other layers unit test

**Comment by iskakaushik** [comment]
Sorry for the delay on this, the test seems to be checking that the contents don't match in case checkerboard is enabled. This is definitely an improvement over whatever testing existed.

I think it's acceptable for this PR to have this, I'll follow-up to see if theres a way to assert that checkerboard indeed exists. It might make sense to move these to their own test class, `checkerboard_layertree_tests.cc` as they tend to be quite verbose.

**Comment by fredlee12345678** [comment]
@iskakaushik I have updated all test case according to your comments, if these is no problem, pls help me to merge this PR. Thanks very much!

**Pull request merged**